### DISCUSSION TITLE
How to we get exception description from IotHubException

### DISCUSSION THREAD

**Discussion by ranjith-t** [Azure/azure-iot-sdk-java#1618] [Q&A]
Hi,
 while we get IotHubException for some reason, we get the errorCode and the errorCodeDescription, but they look to be modified by SDK. How we can get the actual Hub error message?

For example:

catch(ex:IotHubException) {
ex.errorCode and ex.errorCodeDescription // **gives -> errorCode: 409, errorCodeDescription : Conflict**
}

But when we print the exception ex, we get
"
**Exception IoTHub Error: com.microsoft.azure.sdk.iot.service.exceptions.IotHubConflictException: ErrorCode:DeviceAlreadyExists;A device with ID 'new-device-from-ui-1' is already registered. Tracking ID:xxxxxxxxxxxxxxxxxxxxxx-G:0-TimeStamp:10/19/2022 05:12:00**
"

Now, I would like to get the 
- **errorcode -> "DeviceAlreadyExists"** and 
- **description -> "A device with ID 'new-device-from-ui-1' is already registered. Tracking ID:xxxxxxxxxxxxxxxxxxxxxx-G:0-TimeStamp:10/19/2022 05:12:00"**

is there a API to get them? Using String on it does not feel like a good option, as it can change in future?

**Comment by timtay-microsoft** [comment]
When the SDK throws an exception like this, the service's error message is saved as the thrown exception's ```message``` field. See below for more details:

```java
try
{
    registryClient.getDevice("someDeviceId");
}
catch (IotHubException e)
{
    //error code from the service. For example, 404
    System.out.println(e.getErrorCode());

    //human readable error code derived from the integer status code above. For example, "NotFound"
    System.out.println(e.getErrorCodeDescription());

    // human readable error message from the service. For example: "The requested device could not be found. Tracking ID: _______"
    System.out.println(e.getMessage());
}
```

**Reply by ranjith-t** [reply]
OK, I tried getMessage() API, but I was getting reference error. That is the reason I asked this qn, sorry I didn't include that bit my initial qn.

 **Unresolved reference: getMessage**
